wallet/src/de/schildbach/wallet/service/platform/IdentityRepository.kt

```
@@ -0,0 +1,901 @@
+    override suspend fun getLocalUserProfile(): DashPayProfile? {
+        val blockchainIdentityBaseData = loadBlockchainIdentityBaseData()!!
+        return dashPayProfileDao.loadByUserId(blockchainIdentityBaseData.userId!!)
+    }
+
+    override fun getIdentityFromPublicKeyId(): Identity? {
+        return try {
+            platformRepo.getWalletEncryptionKey()?.let {
+                val firstIdentityKey = platformRepo.getBlockchainIdentityKey(0, it) ?: return null
+                platform.stateRepository.fetchIdentityFromPubKeyHash(firstIdentityKey.pubKeyHash)
+            }
+        } catch (e: MaxRetriesReachedException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)